### PR TITLE
EZP-31837: Creating an additional Content Location does not repect Content visibility

### PR DESCRIPTION
**Pull request by pspanja**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | [EZP-31837](https://jira.ez.no/browse/EZP-31837)
| **Type**                                   | bug
| **Target eZ Platform version** | `v7.5`
| **BC breaks**                          | no
| **Tests pass**                          | yes
| **Doc needed**                       | no

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

**Review by ViniTou — APPROVED** [review]

**Review by adamwojs — APPROVED** [review]

File: eZ/Publish/API/Repository/Tests/LocationServiceTest.php [hunk 1]
```diff
@@ -130,6 +130,50 @@ public function testCreateLocation()
         ];
     }
 
+    /**
+     * Test for the createLocation() method.
+     *
+     * @see \eZ\Publish\API\Repository\LocationService::createLocation()
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/LocationServiceTest.php]: Nowadays we rather try to do:
```suggestion
     * @covers \eZ\Publish\API\Repository\LocationService::createLocation
```

File: eZ/Publish/API/Repository/Tests/LocationServiceTest.php [hunk 2]
```diff
@@ -130,6 +130,50 @@ public function testCreateLocation()
...
+     * @depends eZ\Publish\API\Repository\Tests\LocationServiceTest::testCreateLocation
+     * @depends eZ\Publish\API\Repository\Tests\ContentServiceTest::testHideContent
+     */
+    public function testCreateLocationChecksContentVisibility()
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/LocationServiceTest.php]: CS
```suggestion
    public function testCreateLocationChecksContentVisibility(): void
```

File: eZ/Publish/API/Repository/Tests/LocationServiceTest.php [hunk 3]
```diff
@@ -130,6 +130,50 @@ public function testCreateLocation()
...
+        $location = $locationService->createLocation(
+            $contentInfo,
+            $locationCreate
+        );
+        /* END: Use Case */
+
+        $this->assertInstanceOf(
+            '\\eZ\\Publish\\API\\Repository\\Values\\Content\\Location',
+            $location
+        );
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/LocationServiceTest.php]: Let's use FQCN instead to support usage search and let's also abstain from calling $this on static methods, per the latest EA extended inspections recommendations:
```suggestion
        self::assertInstanceOf(
            Location::class,
            $location
        );
```

File: eZ/Publish/API/Repository/Tests/LocationServiceTest.php [hunk 4]
```diff
@@ -130,6 +130,50 @@ public function testCreateLocation()
...
+
+        $this->assertTrue($location->invisible);
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/LocationServiceTest.php]: as above:
```suggestion
        self::assertTrue($location->invisible);
```
yes I know, we should run rector on the entire package ;)

**Review by alongosz — CHANGES_REQUESTED** [review]
Looks good, minor CS remarks:

File: eZ/Publish/API/Repository/Tests/LocationServiceTest.php (hunk 1 above)
Comment (pspanja) [reply eZ/Publish/API/Repository/Tests/LocationServiceTest.php]: Fixed in https://github.com/ezsystems/ezpublish-kernel/pull/3066/commits/c5536b0a3ec34bbdc63a99ecd01b0cbb74705187

File: eZ/Publish/API/Repository/Tests/LocationServiceTest.php (hunk 2 above)
Comment (pspanja) [reply eZ/Publish/API/Repository/Tests/LocationServiceTest.php]: Fixed in https://github.com/ezsystems/ezpublish-kernel/pull/3066/commits/c5536b0a3ec34bbdc63a99ecd01b0cbb74705187

File: eZ/Publish/API/Repository/Tests/LocationServiceTest.php (hunk 3 above)
Comment (pspanja) [reply eZ/Publish/API/Repository/Tests/LocationServiceTest.php]: Fixed in https://github.com/ezsystems/ezpublish-kernel/pull/3066/commits/c5536b0a3ec34bbdc63a99ecd01b0cbb74705187

File: eZ/Publish/API/Repository/Tests/LocationServiceTest.php (hunk 4 above)
Comment (pspanja) [reply eZ/Publish/API/Repository/Tests/LocationServiceTest.php]: You should also update CS rules ;)

Fixed in https://github.com/ezsystems/ezpublish-kernel/pull/3066/commits/c5536b0a3ec34bbdc63a99ecd01b0cbb74705187

**Review by alongosz — APPROVED** [review]

**Review by micszo — APPROVED** [review]
QA Approved on eZ Platform EE v2.5.14 & v3.1.3 with diff with PAPI.

**Comment by lserwatka** [comment]
@alongosz could you merge it up?

**Comment by alongosz** [comment]
Merged into `ezplatform-kernel:1.1` as ezsystems/ezplatform-kernel@42fee313e0d30cce0e6a300c2b96717f94cf00d8,  via ezsystems/ezplatform-kernel@46bd2ea9dc02b83c3f31d624024ea60bf1a3f8b1 and up to `ezplatform-kernel:master` via ezsystems/ezplatform-kernel@8c8e2873c644c35b16402b88b15af3b75eedaaa3.

Thank you @pspanja :tada:

**Pull request merged**